packages/yew/src/dom_bundle/braw.rs

```diff
@@ -0,0 +1,220 @@
+use wasm_bindgen::JsCast;
+use web_sys::Element;
+
+use crate::dom_bundle::bnode::BNode;
+use crate::dom_bundle::traits::{Reconcilable, ReconcileTarget};
+use crate::dom_bundle::utils::insert_node;
+use crate::dom_bundle::BSubtree;
+use crate::html::AnyScope;
+use crate::virtual_dom::VRaw;
+use crate::NodeRef;
+
+#[derive(Debug)]
+pub struct BRaw {
+    reference: NodeRef,
+    children_count: usize,
+}
+
+impl BRaw {
+    fn create_elements(html: &str) -> Vec<Element> {
+        let div = gloo::utils::document().create_element("div").unwrap();
+        let html = html.trim();
+        div.set_inner_html(html);
+        let children = div.children();
+        let children = js_sys::Array::from(&children);
+        let children = children.to_vec();
+        children
+            .into_iter()
+            .map(|it| it.unchecked_into())
+            .collect::<Vec<_>>()
+    }
+
+    fn detach_bundle(&self, parent: &Element) {
+        if let Some(node) = self.reference.cast::<Element>() {
+            let mut next_sibling = node.unchecked_ref::<Element>().next_sibling();
+
+            parent
+                .remove_child(&node)
+                .expect("failed to remove braw node");
+            let len = self.children_count - 1;
+            for _ in 0..len {
+                if let Some(node) = next_sibling {
+                    next_sibling = node.clone().unchecked_ref::<Element>().next_sibling();
+                    parent
+                        .remove_child(&node)
+                        .expect("failed to remove braw node");
+                }
+            }
+        }
+    }
+}
+
+impl ReconcileTarget for BRaw {
+    fn detach(self, _root: &BSubtree, parent: &Element, _parent_to_detach: bool) {
+        self.detach_bundle(parent);
+    }
+
+    fn shift(&self, next_parent: &Element, next_sibling: NodeRef) -> NodeRef {
+        if let Some(node) = self.reference.cast::<Element>() {
+            if let Some(parent) = node.parent_node() {
+                parent.remove_child(&node).unwrap();
+            }
+
+            next_parent
+                .insert_before(&node, next_sibling.get().as_ref())
+                .unwrap();
```

Comment:
Seems to have the same issue as detach where it only moves the first element. Moreover, you don't need to explicitly remove the nodes, this happens implicitly when they are inserted as children of another node. That should save some calls out to DOM:

> If the given node already exists in the document, insertBefore() moves it from its current position to the new position. (That is, it will automatically be removed from its existing parent before appending it to the specified new parent.)
- [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore)

Reply:
> Seems to have the same issue as detach where it only moves the first element.

How exactly would that be fixed here?

Reply:
Actually, I'm not even sure what the best way to test this is. Current ideas:
(1) one could manually call the shift implementation or
(2) one could have (this depends too much on the diffing algorithm, imo)
- a component `<Comp />` that returns `Html::from_raw_html` from its view function
- use that component as part of a keyed list and in a re-render swap it with another element in that list
(3) have a portal and point that portal to a new host as part of a re-render. This should work in the future, but depends on a correctly working portal

In any case, to answer the actual question, I'd iterate [like here](https://github.com/yewstack/yew/pull/2842/files#diff-17c9c939b65a8145eda52bb394810f45f401186ef3c89816cb8a4c772fc45f5bR34-R40) over the elements.